srml/support/src/traits.rs

```diff
@@ -65,10 +65,131 @@ pub trait ArithmeticType {
 	type Type: SimpleArithmetic + As<usize> + As<u64> + Codec + Copy + MaybeSerializeDebug + Default;
 }
 
+/// Simple trait designed for hooking into a transaction payment.
+///
+/// It operates over a single generic `AccountId` type.
+pub trait MakePayment<AccountId> {
+	/// Make transaction payment from `who` for an extrinsic of encoded length
+	/// `encoded_len` bytes. Return `Ok` iff the payment was successful.
+	fn make_payment(who: &AccountId, encoded_len: usize) -> Result<(), &'static str>;
+}
+
+impl<T> MakePayment<T> for () {
+	fn make_payment(_: &T, _: usize) -> Result<(), &'static str> { Ok(()) }
+}
+
+/// Handler for when some currency "account" decreased in balance for
+/// some reason.
+///
+/// The only reason at present for an increase would be for validator rewards, but
+/// there may be other reasons in the future or for other chains.
+///
+/// Reasons for decreases include:
+///
+/// - Someone got slashed.
+/// - Someone paid for a transaction to be included.
+pub trait OnUnbalanced<Imbalance> {
+	/// Handler for some imbalance. Infallible.
+	fn on_unbalanced(amount: Imbalance);
+}
+
+impl<Imbalance: Drop> OnUnbalanced<Imbalance> for () {
+	fn on_unbalanced(amount: Imbalance) { drop(amount); }
+}
+
+/// Simple boolean for whether an account needs to be kept in existence.
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub enum ExistenceRequirement {
+	/// Operation must not result in the account going out of existence.
+	KeepAlive,
+	/// Operation may result in account going out of existence.
+	AllowDeath,
+}
+
+/// A trait for a not-quite Linear Type that tracks an imbalance.
+///
+/// Imbalances can either be Positive (funds were added somewhere without
+/// being subtracted elsewhere - e.g. a reward) or Negative.(funds deducted
```

Comment:
nit: remove period here: or Negative.(funds

Reply:
Hey @joepetrowski , there is a really helpful feature called "suggestions". You can send it and the author can apply it right away without an editor-commit-push round-trip .

```suggestion
/// being subtracted elsewhere - e.g. a reward) or Negative (funds deducted
```

Reply:
Oh thanks @pepyakin I see how to do it